pages/lazer/how-lazer-works.mdx

```diff
@@ -1,5 +1,77 @@
+import { Callout } from "nextra/components";
+
 # How Pyth Lazer works
 
-Pyth Lazer is a permissioned service that provides ultra-low-latency price and market data to highly latency-sensitive users.
+Pyth Lazer is a permissioned service that provides ultra-low-latency market data to consumers.
+It aggregates data from multiple publishers and distributes it to consumers through a multi-tier architecture.
+
+## System Services
+
+The architecture consists of five main services that work together to provide ultra-low-latency data to consumers.
+
+### INSERT DIAGRAM HERE
+
+### Publishers
+
+Publishers are the entities that provide market data to Lazer. They submit updates via authenticated WebSocket connections.
+Each publisher is configured with specific permissions defining which feeds they can update.
+
+### Relayers
+
+The Relayer service is the ingestion layer that receives and validates all incoming updates from publishers.
+
+**Key responsibilities:**
+
+- **Authentication**: Validates publisher access tokens and optional Ed25519 signatures.
+- **Rate limiting**: Enforces configurable limits on publisher updates.
+- **Message forwarding**: Publishes validated updates to an internal message queue (a NATS cluster).
+
+**Douro Labs** operates the relayer service for the Pyth Lazer network.
+
+<Callout type="info">
+  The Douro Labs-operated relayer follows a strict, deterministic processing model:
+  - **No price dropping**: All validated updates are forwarded to the message queue without dropping any prices.
+  - **FCFS processing**: Updates are processed on a first-come-first-served basis without prioritization.
+  - **Deterministic operation**: The relayer operates according to its configured logic and does not deviate from it.
+
+This ensures reliable, predictable data flow from publishers to consumers.
+
+</Callout>
+
+### Message Queue
+
+The system uses NATS JetStream as the primary message queue for pub/sub messaging with stream persistence.
+This allows the system to be deployed in a multi-datacenter environment and ensures reliable message delivery between services.
+
+**Message ordering**: NATS JetStream guarantees message ordering within a single stream, ensuring that updates from publishers are processed in the order they were received by the Relayer.
```

Comment:
As I mentioned above, I would avoid specifying implementation details and focus more on what this part does -> perform consensus, ensure unified ordering, etc

Reply:
yeah agree with this comment. don't use the word "NATS", just say "message queue"